integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ClusterTemplateTests.java

```diff
@@ -0,0 +1,245 @@
+package com.sequenceiq.it.cloudbreak;
+
+import com.sequenceiq.it.cloudbreak.newway.CloudbreakClient;
+import com.sequenceiq.it.cloudbreak.newway.CloudbreakTest;
+import com.sequenceiq.it.cloudbreak.newway.ClusterTemplate;
+import com.sequenceiq.it.cloudbreak.newway.TestParameter;
+import com.sequenceiq.it.cloudbreak.newway.cloud.CloudProvider;
+import com.sequenceiq.it.cloudbreak.newway.cloud.CloudProviderHelper;
+import com.sequenceiq.it.cloudbreak.newway.cloud.OpenstackCloudProvider;
+import com.sequenceiq.it.util.LongStringGeneratorUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StreamUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+
+public class ClusterTemplateTests extends CloudbreakTest {
+
+    private static final String VALID_CT_NAME = "valid-template";
+
+    private static final String AGAIN_CT_NAME = "again-template";
+
+    private static final String LONG_DC_CT_NAME = "long-description-template";
+
+    private static final String DELETE_CT_NAME = "delete-template";
+
+    private static final String SPECIAL_CT_NAME = "@#$|:&* ABC";
+
+    private static final String ILLEGAL_CT_NAME = "Illegal template name %s";
+
+    private static final String INVALID_SHORT_CT_NAME = "";
+
+    private static final String EMPTY_CT_NAME = "temp-empty-template";
+
+    private static final String INVALIDURL_CT_NAME = "temp-url-template";
+
+    private static final String CT_DESCRIPTION = "temporary template for API E2E tests";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClusterTemplateTests.class);
+
+    private String errorMessage = "";
+
+    private CloudProvider cloudProvider;
+
+    @Inject
+    private LongStringGeneratorUtil longStringGeneratorUtil;
+
+    public ClusterTemplateTests() {
+    }
+
+    public ClusterTemplateTests(CloudProvider cp, TestParameter tp) {
+        cloudProvider = cp;
+        setTestParameter(tp);
+    }
+
+    @Test(priority = 1, groups = "templates")
+    public void testCreateValidClusterTemplate() throws Exception {
+        given(CloudbreakClient.created());
```

Comment:
I don't know where to put this since my comment concerns more places in the test so I put this here. 😃 
As I can see many of the test cases differ only in some key component such as an invalid character or template name but basically the tests are the same.
We can make a data provider for these tests where the difference between two test is not in the execution type but in one of it's components.

For example in this case: [link](https://github.com/hortonworks/cloudbreak/pull/4083/files#diff-683605b85b437b29d1ebf205b5e3c215R146)

Reply:
I've introduced a new data provider for template name tests:
```
@Test(expectedExceptions = BadRequestException.class, priority = 2, dataProvider = "templatename", groups = "templates")
    public void testCreateInvalidNameClusterTemplate(TestContext testContext, String templateName) throws Exception {
```